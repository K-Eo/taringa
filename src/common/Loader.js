import React from 'react'
import classNames from 'classnames'

const Loader = ({ className, size }) => {
  const classes = classNames('text-center text-dark', className)
  const icon = classNames('fa fa-spinner fa-spin', size)

  return (
    <div className={classes}>
      <i className={icon} />
    </div>
  )
}

export default Loader
