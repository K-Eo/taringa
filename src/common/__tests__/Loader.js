import React from 'react'

import Loader from '../Loader'

describe('Loader', () => {
  it('exists', () => {
    expect(Loader).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Loader />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with classes', () => {
    const wrapper = shallow(<Loader className="my-4" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with different size', () => {
    const wrapper = shallow(<Loader size="fa-2x" />)

    expect(wrapper).toMatchSnapshot()
  })
})
