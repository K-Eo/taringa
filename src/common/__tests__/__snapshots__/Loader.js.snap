// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders default 1`] = `
<div
  className="text-center text-dark"
>
  <i
    className="fa fa-spinner fa-spin"
  />
</div>
`;

exports[`Loader renders with classes 1`] = `
<div
  className="text-center text-dark my-4"
>
  <i
    className="fa fa-spinner fa-spin"
  />
</div>
`;

exports[`Loader renders with different size 1`] = `
<div
  className="text-center text-dark"
>
  <i
    className="fa fa-spinner fa-spin fa-2x"
  />
</div>
`;
